src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/GerritServer.java

```diff
@@ -914,14 +915,21 @@ private void rename(String newName) {
      * @param oldName the old name of the Gerrit server
      */
     private void changeSelectedServerInJobs(String oldName) {
-        for (AbstractProject job : PluginImpl.getConfiguredJobs_(oldName)) {
-            GerritTrigger trigger = (GerritTrigger)job.getTrigger(GerritTrigger.class);
+        for (Job job : PluginImpl.getConfiguredJobs_(oldName)) {
+
+            // TODO: find a way to handle add/remove of triggers
+            if (!(job instanceof AbstractProject)) {
+                return;
```

Comment:
you can add a trigger easily. It seems like the trigger add code will remove the old trigger already. Sadly WorkflowJob.java does not export a remove trigger function like the code does. Maybe we just don't bother removing the old trigger at all when a server is removed?

What's the reason that we actually have remove/add trigger code here specifically? I presume to prevent a bug? What do other trigger plugins do to prevent this sort of issue?